### PR TITLE
Add ability to specify path to OIDC token

### PR DESCRIPTION
**Pull request by mpminardi**
Adds the ability to specify a path to an OIDC token in place of supplying the contents of the token.

### CODE REVIEW

File: authentication/auth_method_oidc.go [hunk 1]
```diff
@@ -65,19 +68,39 @@ func (a oidcAuth) getMSALToken(ctx context.Context, api environments.Api, _ auto
 		return &authWrapper.Authorizer{Authorizer: conf.TokenSource(ctx)}, nil
 	}
 
+	idToken := a.idToken
+
+	if a.idTokenPath != "" {
```
Comment (mpminardi) [inline authentication/auth_method_oidc.go:73]: Not sure on priority of the token path vs actual token here, or if we would want to error out when both are supplied instead.

File: authentication/auth_method_oidc.go [hunk 2]
```diff
@@ -27,6 +29,7 @@ func (a oidcAuth) build(b Builder) (authMethod, error) {
 		clientId:            b.ClientID,
 		environment:         b.Environment,
 		idToken:             b.IDToken,
+		idTokenPath:         b.IDTokenPath,
```
Comment (katbyte) [inline authentication/auth_method_oidc.go:32]:
```suggestion
		idTokenFilePath:         b.IDTokenFilePath,
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @mpminardi - couple minor comments i've left inline but otherwise looks good

File: authentication/auth_method_oidc.go [hunk 3]
```diff
@@ -65,19 +68,43 @@ func (a oidcAuth) getMSALToken(ctx context.Context, api environments.Api, _ auto
...
+
+	if a.idTokenFilePath != "" {
+		idToken, err = a.readTokenFile(a.idTokenFilePath)
+
+		if err != nil {
+			return nil, fmt.Errorf("reading token file: %v", err)
+		}
+
+		if a.idToken != "" && a.idToken != idToken {
+			return nil, fmt.Errorf("mismatch between supplied OIDC token and supplied OIDC token file")
```
Comment (mpminardi) [inline authentication/auth_method_oidc.go:81]: @katbyte have added an error when the token doesn't match what is read from the token file based on your recommendation in https://github.com/hashicorp/go-azure-helpers/pull/131#discussion_r980315532. Let me know if you think this text should be adjusted at all!

Comment (katbyte) [reply authentication/auth_method_oidc.go:81]: would it be a good idea to output the tokens?

Comment (katbyte) [inline authentication/auth_method_oidc.go:81]: ie
```suggestion
			return nil, fmt.Errorf("mismatch between supplied OIDC token (%s) and supplied OIDC token file contents (%s) - please either remove one or ensure they match", a.idToken, idToken)
```

Comment (manicminer) [reply authentication/auth_method_oidc.go:81]: I'm not sure that we should to be honest - we go to pains to exclude tokens from log output in other places.

**Review by manicminer — APPROVED** [review]
Thanks for the contribution and changes @mpminardi, this LGTM!

**Pull request merged**